packages/python/plotly/plotly/express/_core.py

```diff
@@ -1955,10 +1955,37 @@ def make_figure(args, constructor, trace_patch=None, layout_patch=None):
     trace_specs, grouped_mappings, sizeref, show_colorbar = infer_config(
         args, constructor, trace_patch, layout_patch
     )
-    grouper = [x.grouper or one_group for x in grouped_mappings] or [one_group]
-    grouped = args["data_frame"].groupby(grouper, sort=False)
+    grouper_ = [x.grouper or one_group for x in grouped_mappings] or [one_group]
+
+    all_same_group = True  # variable indicating if grouping can be avoided
+    for g in grouper_:
+        if g is not one_group:
+            arr = args["data_frame"][g].values
+            all_same_group &= (arr[0] == arr).all(axis=0)
+        if not all_same_group:
+            break  # early stopping if not all the same
+
+    if all_same_group:
+        # Do not perform an expensive groupby operation when there are either
+        # no groups to group by, or when the group has only one (i.e., the same) value
+        grouper = [g for g in grouper_ if g is not one_group]
+        assert len(grouper) <= 1
+        # -> create orders, sorted_group_names equivalent to those of get_ordings
```

Comment:
I love the optimization where we don't group if the groupers are all `one_group` but I'm worried about the duplication of logic here... If we just do the optimization when no grouping is requested at all (i.e. leave out the case when we do have e.g. `symbol` but there's only one value) then this PR can be less invasive/have less duplication. I've done an implementation of that over here https://github.com/plotly/plotly.py/pull/3765 if you want to take a look at it.

I think that most of the value of this PR comes from the "all `one_group`" case, and that leaving out the case where there happens to only be one group would be OK for now as this is likely not all that common, but I'm not dead set on it.